qiskit/pulse/__init__.py

```diff
@@ -130,15 +145,98 @@
 
 """
 
-from .channels import (DriveChannel, MeasureChannel, AcquireChannel,
-                       ControlChannel, RegisterSlot, MemorySlot)
-from .commands import AcquireInstruction, FrameChange, PersistentValue
-from .configuration import LoConfig, LoRange, Kernel, Discriminator
-from .exceptions import PulseError
-from .instruction_schedule_map import InstructionScheduleMap
-from .instructions import Acquire, Instruction, Delay, Play, ShiftPhase, Snapshot, SetFrequency
-from .interfaces import ScheduleComponent
-from .pulse_lib import (SamplePulse, Gaussian, GaussianSquare, Drag,
-                        Constant, ConstantPulse, ParametricPulse)
-from .pulse_lib.samplers.decorators import functional_pulse
-from .schedule import Schedule
+# Builder imports.
+from qiskit.pulse.builder import (
+    # builder construction methods.
+    active_backend,
+    active_transpiler_settings,
+    active_circuit_scheduler_settings,
+    append_instruction,
+    build,
+    num_qubits,
+    qubit_channels,
+    samples_to_time,
+    time_to_samples,
+    # builder instructions.
+    acquire,
+    barrier,
```

Comment:
I found something when using barriers which may be a bug. 

A barrier taking a single argument should be ignored. Barriers really make sense when they act on 2 or more channels (or 2 or more qubits), as it syncs those locations in time.

But in the example below, the barrier is pushing operations back. This example works as expected if you move the barrier two lines up to be outside the second `with` scope.
```python
from qiskit import pulse
d0 = pulse.DriveChannel(0)
d1 = pulse.DriveChannel(1)
d2 = pulse.DriveChannel(2)

drag_1 = pulse.Drag(100, 1.0, 15.0, 0.1)
drag_2 = pulse.Drag(200, 1.0, 30.0, 0.1)
drag_3 = pulse.Drag(300, 1.0, 50.0, 0.1)

with pulse.build() as sched:
    pulse.play(drag_1, d0)
    with pulse.align_left():
        pulse.barrier(d0)
        pulse.play(drag_2, d1)
        pulse.play(drag_3, d2)

sched.draw()
```
![image](https://user-images.githubusercontent.com/8622381/81453861-a33c2980-9158-11ea-9ac8-5630e9f9ef79.png)

Reply:
> A barrier taking a single argument should be ignored. Barriers really make sense when they act on 2 or more channels (or 2 or more qubits), as it syncs those locations in time.

This is not true in the case of qubits, as this decomposes into channels that must be barriers. A single channel barrier should be ignored and I think it will be.

Reply:
Fixed in fe27eb617